packs/loaders/productListiningPage.ts

```diff
@@ -18,52 +18,47 @@ interface PLPPageParams {
   productApiProps: Partial<
     Omit<
       GetProductProps,
-      | "id"
-      | "idColecaoProdutos"
-      | "offset"
-      | "somenteCronometrosAtivos"
-      | "ordenacao"
       | "url"
       | "page"
+      | "offset"
+      | "ordenacao"
     >
   >;
   plpProps: Omit<ProductListiningPageProps, "productsData" | "baseURL">;
 }
 
+type Props = Omit<
+  GetProductProps,
+  "IdCategoria" | "IdSubCategoria" | "url" | "page"
+>;
+
 /**
- * @title Otica Isabela Products Listining Page
+ * @title Otica Isabela Dias - Product Listining Page
  * @description Works on routes /busca using the querystring "termo" to serch the products OR in categories pages on routes /category
  */
 const loaders = async (
-  _props: null,
+  props: Props,
   req: Request,
   ctx: Context,
 ): Promise<ProductListingPage | null> => {
   const { configStore: config } = ctx;
   const url = new URL(req.url);
   const path = paths(config!);
+  const { offset } = props;
 
   const isCategoryPage = !url.pathname.includes("busca");
 
   const pageParams: PLPPageParams | null = isCategoryPage
-    ? await getCategoryPageParams(url, config!).then((data) => data)
-    : {
-      productApiProps: {
-        nome: url.searchParams.get("termo") ?? "",
-      },
-      plpProps: {
-        term: url.searchParams.get("termo") ?? "",
-        pageType: "search",
-      },
-    };
+    ? await getCategoryPageParams(url, config!, props).then((data) => data)
+    : getSearchPageParams(url, props);
 
   if (!pageParams) return null;
 
   const products = await fetchAPI<ProductData>(
     path.product.getProduct({
-      offset: 32,
+      offset: offset ?? 32,
```

Comment:
offset tem que ser obrigatorio.
```suggestion
      offset,
```